app/controllers/restaurants_controller.rb

```diff
@@ -1,10 +1,13 @@
 class RestaurantsController < ApplicationController
-  def index
-    @restaurants = Restaurant.all
-  end
 
   def show
     @restaurant = Restaurant.find(params[:id])
     @menus = @restaurant.menus.all
   end
+
+  def search
```

Comment:
I would like you to talk me through this implementation.

Reply:
Adding the sad path and taking care of the other requests, then we can talk about it